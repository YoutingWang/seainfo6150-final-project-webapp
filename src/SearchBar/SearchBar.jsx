import React, {useState} from 'react';
import styles from "./SearchBar.module.css";

const SearchBar = (props) => {
    const [term, setTerm] = useState(props.term || '');
    const [location, setLocation] = useState(props.location || '');

    function submit(e) {
        if(typeof props.search === 'function') {
            props.search(term, location);
        }
        e.preventDefault();
    }

    return (
        <form onSubmit={submit}>
            <div className={styles.textBox}>
            <div>
                    <p className={styles.text}>Name</p>
            </div>
                <p>
                    <input className={styles.input}
                           onChange={(e) => setTerm(e.target.value)}
                           type="text"
                           value={term}
                           placeholder="pizza, chinese..."
                    />
                </p>
                <div>
                    <p className={styles.text}>Location</p>
                </div>
                <p>
                    <input className={styles.input}
                           onChange={(e) => setLocation(e.target.value)}
                           type="text"
                           placeholder="where..."/>
                </p>
                <div onClick={submit}>
                </div>
                <p>
                    <button className={styles.button}>Search</button>
                </p>
            </div>
        </form>
    );
}

export default SearchBar;