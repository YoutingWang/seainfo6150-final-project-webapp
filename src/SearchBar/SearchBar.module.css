.input {
    min-width: 80px;
}

.textBox{
    display: flex;
    flex-direction: row;
    justify-content: center;
}

.text {
    font-size: 13px;
    margin-top: 16px;
    margin-left: 12px;
    margin-right: 12px;
    font-family: 'Raleway', sans-serif;
}

.button{
    font-size: 13px;
    font-family: 'Nunito', sans-serif;
}

@media screen and (min-width: 550px) {
    .input {
        min-width: 80px;
    }
    .text {
        font-size: 15px;
    }
    .button {
        font-size: 13px;
    }
}

@media screen and (min-width: 700px) {
    .input {
        min-width: 100px;
    }
    .text {
        font-size: 18px;
    }
    .button {
        font-size: 14px;
    }
}

@media screen and (min-width: 900px) {
    .input {
        min-width: 120px;
    }
    .text {
        font-size: 20px;
    }
    .button {
        font-size: 15px;
    }
}