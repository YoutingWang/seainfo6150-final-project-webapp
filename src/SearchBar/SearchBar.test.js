import React from "react";
import {
    render
} from "@testing-library/react";
import SearchBar from "./SearchBar.jsx";

const search = {
    "term": "pizza",
    "location": "san jose"
};

it("renders the home component correctly", () => {
            const {
                container
            } = render( < SearchBar search = {
                    search
                }
                / > );
                expect(container).toMatchSnapshot();
            });