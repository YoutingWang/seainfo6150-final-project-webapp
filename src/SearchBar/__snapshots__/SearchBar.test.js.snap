// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the home component correctly 1`] = `
<div>
  <form>
    <div
      class="textBox"
    >
      <div>
        <p
          class="text"
        >
          Name
        </p>
      </div>
      <p>
        <input
          class="input"
          placeholder="pizza, chinese..."
          type="text"
          value=""
        />
      </p>
      <div>
        <p
          class="text"
        >
          Location
        </p>
      </div>
      <p>
        <input
          class="input"
          placeholder="where..."
          type="text"
        />
      </p>
      <div />
      <p>
        <button
          class="button"
        >
          Search
        </button>
      </p>
    </div>
  </form>
</div>
`;
